modules/apps/08-callbacks/ics20_example_cb.go

```diff
@@ -0,0 +1,58 @@
+package callbacks
+
+import (
+	"encoding/json"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/ibc-go/v7/modules/apps/transfer/types"
+	channeltypes "github.com/cosmos/ibc-go/v7/modules/core/04-channel/types"
+)
+
+// MyICS20Payload my random payload example
+type MyICS20Payload struct {
+	SrcCallbackAddress string `json:"src_callback_address"`
+	DstCallbackAddress string `json:"dst_callback_address"`
+}
+
+var _ Decoder[MyICS20Payload] = &MyICS20Decoder{}
+
+type MyICS20Decoder struct { //nolint:gofumpt
+}
+
+func (m MyICS20Decoder) Decode(packet channeltypes.Packet) (*MyICS20Payload, error) {
+	var data types.FungibleTokenPacketData
+	if err := types.ModuleCdc.UnmarshalJSON(packet.GetData(), &data); err != nil {
+		return nil, err
+	}
+	// do we need a sanity check that type was unpacked proper? like denom not empty
+
+	if data.Memo == "" {
+		return nil, nil
+	}
+	// unmarshal json to type, I prefer using concrete type here:
+	var r MyICS20Payload
+	if err := json.Unmarshal([]byte(data.Memo), &r); err != nil {
+		return nil, err
+	}
+	// todo: check that contains valid data
+	return &r, nil
+}
+
+var _ Executor[MyICS20Payload] = &MyICS20Callbacks{}
```

Comment:
Executor for ics20 only. my assumption was that chains may want to customize the callbacks for each of the ibc apps.